ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -617,7 +617,15 @@ public void handleReplicaMetadataRequest(Request request) throws IOException, In
             ReplicaMetadataResponseInfo replicaMetadataResponseInfo =
                 new ReplicaMetadataResponseInfo(partitionId, findInfo.getFindToken(), findInfo.getMessageEntries(),
                     remoteReplicaLagInBytes);
+            if (replicaMetadataResponseInfo.getTotalSizeOfAllMessages()
+                > 5 * replicaMetadataRequest.getMaxTotalSizeOfEntriesInBytes()) {
+              logger.debug("{} generated a metadata response {} where the cumulative size of messages is {}",
+                  replicaMetadataRequest, replicaMetadataResponseInfo,
+                  replicaMetadataResponseInfo.getTotalSizeOfAllMessages());
+              metrics.replicationResponseMessageSizeTooHigh.inc();
+            }
             replicaMetadataResponseList.add(replicaMetadataResponseInfo);
+            metrics.replicaMetadataTotalSizeOfMessages.update(replicaMetadataResponseInfo.getRemoteReplicaLagInBytes());
```

Comment:
Shouldn't this be getTotalSizeOfAllMessages()?

Reply:
Yup - thanks for catching it !